README.md

```diff
@@ -1,2 +1,145 @@
+          .port(9052)
+          .common_tags({{"service", "statsdex"}, {"env", "production"}})
+          .Build();
+
+  // Create a scope which serves as a registry for your metrics.
+  auto scope = tally::ScopeBuilder().reporter(reporter).ReportEvery(
+      std::chrono::seconds(1));
+
+  // Create your metrics.
+  auto counter = scope->Counter("my_counter");
+  auto gauge = scope->Gauge("my_gauge");
+  auto timer = scope->Timer("my_timer");
+  auto histogram = scope->Histogram("my_histogram",
+                                    tally::Buckets::LinearValues(0.0, 1.0, 10));
+
+  for (;;) {
+    // Update your metrics.
+    counter->Inc(1);
+    gauge->Update(5.0);
+    timer->Record(std::chrono::nanoseconds(3000000));
+    histogram->Record(8.0);
+
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+  }
+}
+```
+
+## Building
+
+[Bazel](https://bazel.build/) is used to build `tally-cpp`. In addition to Bazel, `tally-cpp`
+also requires GNU build tools:
+
+  - autoconf 2.65
+  - automake 1.13
+  - libtool 1.5.24
+
+This requirement stems from the fact that `tally-cpp` depends on
+[Apache thrift](https://github.com/apache/thrift) for the default reporter implementation. Thrift
+uses CMake as its build system and has some pre-build steps which generate necessary header
+files. We use Bazel's [`repository_rules`](https://docs.bazel.build/versions/master/skylark/repository_rules.html)
+feature to ensure these pre-build commands are run after fetching the Thrift source code, following
+the approach taken by Envoy and discussed in
+[this blog post](https://blog.envoyproxy.io/external-c-dependency-management-in-bazel-dd37477422f5).
+
+To use tally in your project you'll need to first at it as a dependency to your project. First
+add the following to your `WORKSPACE` file:
+
+```python
+workspace(name = "com_github_jeromefroe_tally_cpp_example")
+
+http_archive(
+    name = "com_github_m3db_tally_cpp",
+    strip_prefix = "tally-cpp-master",
+    urls = ["https://github.com/m3db/tally-cpp/archive/master.zip"],
+)
+
+load("@com_github_m3db_tally_cpp//:repositories.bzl", "tally_cpp_repositories")
+
+tally_cpp_repositories()
+
+load("@com_github_nelhage_rules_boost//:boost/boost.bzl", "boost_deps")
+
+boost_deps()
+```
+
+When you call `tally_cpp_repositories()` in your WORKSPACE file you'll introduce the following
+dependencies to your project:
+
+  - `load_com_google_googletest()` for Google gtest
+  - `load_com_github_nelhage_rules_boost()` for Boost
+  - `load_org_apache_thrift()` for Apache Thrift
+
+In addition, you also need to load and call `boost_deps()` to add the necessary boost libraries.
+
+Once your `WORKSPACE` file is setup, you can this library as a dependency to your own `BUILD`
+file. For example:
+
+```python
+cc_binary(
+    name = "main",
+    srcs = ["main.cc"],
+    deps = [
+        "@com_github_m3db_tally_cpp//m3",
+        "@com_github_m3db_tally_cpp//tally",
+    ],
+)
+```
+
+## Contributing
```

Comment:
do you want to include a section on building for distribution

Reply:
The `Building` section includes information on how to add the library as a dependency to a Bazel project, anything else you had in mind? As a shared library?

Reply:
usual release management stuff - tagging versions on gh, release notes, etc

Reply:
Gotcha, yea I'll probably add a separate markdown doc for that since it's more internal information so we can keep the README to just client-facing stuff.

Reply:
Sounds good